recipes/fast-cdr/all/conanfile.py

```diff
@@ -0,0 +1,101 @@
+from conans import ConanFile, CMake, tools
+import os
+from conans.tools import load
+from conans.errors import ConanInvalidConfiguration
+from pathlib import Path
+import textwrap
+
+class FastCDRConan(ConanFile):
+
+    name = "fast-cdr"
+    license = "Apache-2.0"
+    homepage = "https://github.com/eProsima/Fast-CDR"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "eProsima FastCDR library for serialization"
+    topics = ("conan","DDS", "Middleware","Serialization")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared":          [True, False],
+        "fPIC":            [True, False]
+    }
+    default_options = {
+        "shared":            False,
+        "fPIC":              True
+    }
+    exports_sources = ["patches/**"]
+    _cmake = None
+
+    @property
+    def _pkg_cmake(self):
+        return os.path.join(
+            self.package_folder,
+            "lib",
+            "cmake"
+        )
+
+    @property
+    def _pkg_share(self):
+        return os.path.join(
+            self.package_folder,
+            "share"
+        )
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def _get_configured_cmake(self):
+        if self._cmake:
+            pass 
+        else:
+            self._cmake = CMake(self)
+        self._cmake.configure(
+            source_dir = self._source_subfolder
+        )
+        return self._cmake
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True,
+                  destination=self._source_subfolder)
+    
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def validate(self):
+        if self.settings.os == "Linux":
+            if (self.settings.compiler == "gcc" and self.settings.compiler.libcxx == "libstdc++"):
```

Comment:
```suggestion
            if (self.settings.compiler == "gcc" and self.settings.compiler.libcxx == "libstdc++"):
```
Why ? Which feature is required? Usually libc++ covers C++ >11 features too.

Reply:
actually this is a strange linker error i currently didnt figure out why. When i do not set the libcxx to libstdc++11 then i get the error : 

"/usr/bin/ld: CMakeFiles/test_package.dir/test_package.cpp.o: in function `eprosima::fastcdr::Cdr::deserialize(std::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> >&)':
/home/kac2st/.conan/data/fast-cdr/1.0.21/_/_/package/9cb7ac99f58a79b591516cbdd53096db4a815c86/include/fastcdr/Cdr.h:2455: undefined reference to `eprosima::fastcdr::Cdr::readWString(unsigned int&)'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/test_package.dir/build.make:85: bin/test_package] Error 1
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/test_package.dir/all] Error 2
make: *** [Makefile:84: all] Error 2"

I was asking myself if it is related to --> https://docs.conan.io/en/1.36/howtos/manage_gcc_abi.html

Reply:
It's probably related to the missing CMake wrapper ([more info](https://github.com/conan-io/conan-center-index/pull/5615#discussion_r638763701)) since it's not changing the settings correctly

Reply:
I give it a try